engine/src/main/java/org/hibernate/validator/spi/nodenameprovider/PropertyNodeNameProvider.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.spi.nodenameprovider;
+
+import org.hibernate.validator.Incubating;
+
+/**
+ * This interface is used to resolve the name of a property node when creating the property path.
+ *
+ * @author Damir Alibegovic
+ * @since 6.1.0
+ */
+@Incubating
+public interface PropertyNodeNameProvider {
+	/**
+	 * Returns resolved name of the property.
+	 * <br><br>
+	 * Depending on the subtype of the {@link Property},
+	 * different strategy for name resolution should be applied, defaulting to {@link Property#getName()}. For example:
```

Comment:
s/different strategy for name resolution should be applied/a different strategy for name resolution could be applied/